core/src/worker/activities.rs

```diff
@@ -360,22 +407,33 @@ impl WorkerActivityTasks {
                             .err()
                     }
                     aer::Status::Cancelled(ar::Cancellation { failure }) => {
-                        let details = if let Some(Failure {
-                            failure_info:
-                                Some(FailureInfo::CanceledFailureInfo(CanceledFailureInfo { details })),
-                            ..
-                        }) = failure
-                        {
-                            details
+                        if matches!(
+                            act_info.issued_cancel_to_lang,
+                            Some(ActivityCancelReason::GracefulShutdown),
+                        ) {
+                            // We don't tell server about cancels for graceful shutdown since they
+                            // were never requested
+                            None
```

Comment:
We currently transform these into failures in lang (mentioned in the tests too), the rationale is that we don't want to wait for timeout until the activity can be restarted.

Reply:
Yeah, makes sense. Forgot about that.